src/components/genes/GenesSubscriber.js

```diff
@@ -1,75 +1,85 @@
-import React, { useState, useEffect, useCallback } from 'react';
+import React, {
+  useState, useEffect, useCallback, useMemo,
+} from 'react';
 import PubSub from 'pubsub-js';
-
-import Genes from './Genes';
-
-import TitleInfo from '../TitleInfo';
 import {
-  GENES_ADD,
-  CELLS_COLOR,
-  CLEAR_PLEASE_WAIT,
-  RESET,
+  EXPRESSION_MATRIX_ADD, CELLS_COLOR, CLEAR_PLEASE_WAIT, RESET,
 } from '../../events';
-import { interpolateColors } from '../utils';
+import { interpolatePlasma } from '../interpolate-colors';
+import { fromEntries, pluralize } from '../utils';
+import TitleInfo from '../TitleInfo';
+import Genes from './Genes';
 
 export default function GenesSubscriber(props) {
   const {
     onReady,
-    mapping,
     removeGridComponent,
-    variablesLabelOverride,
+    variablesLabelOverride: variablesLabel = 'gene',
     theme,
   } = props;
-  const [genes, setGenes] = useState({});
+
+  const [expression, setExpression] = useState();
   const [selectedId, setSelectedId] = useState(null);
   const [urls, setUrls] = useState([]);
 
   const onReadyCallback = useCallback(onReady, []);
 
   useEffect(() => {
-    const genesAddToken = PubSub.subscribe(
-      GENES_ADD, (msg, { data, url }) => {
-        setGenes(data);
-        setUrls((prevUrls) => {
-          const newUrls = [...prevUrls].concat({ url, name: 'Genes' });
-          return newUrls;
+    const expressionMatrixAddToken = PubSub.subscribe(
```

Comment:
I like the precision of `expressionMatrix` because it will allow us to update the schema in the future if gene-expression-specific variables need to be added. For example, it already looks like we will need some way to map the Ensemble gene IDs to the corresponding gene names.